dali/pipeline/operators/op_spec.h

```diff
@@ -426,87 +512,53 @@ inline bool OpSpec::TryGetArgumentImpl(
     auto schema_val = schema.FindDefaultValue(name);
     using VT = const ValueInst<S>;
     if (VT *vt = dynamic_cast<VT *>(schema_val.second)) {
-      result = vt->Get();
+      result = static_cast<T>(vt->Get());
+      return true;
+    }
+  }
+  return false;
+}
+
+
+template <typename T, typename S>
+inline std::vector<T> OpSpec::GetRepeatedArgumentImpl(const string &name) const {
+  using V = std::vector<S>;
+  // Search for the argument locally
+  auto arg_it = arguments_.find(name);
+  if (arg_it != arguments_.end()) {
+    // Found locally - return
+    return detail::convert_vector<T>(arg_it->second->template Get<V>());
+  } else {
+    // Argument wasn't present locally, get the default from the associated schema
+    const OpSchema& schema = SchemaRegistry::GetSchema(this->name());
+    return detail::convert_vector<T>(schema.GetDefaultValueForOptionalArgument<V>(name));
+  }
+}
+
+template <typename T, typename S>
+inline bool OpSpec::TryGetRepeatedArgumentImpl(std::vector<T> &result, const string &name) const {
```

Comment:
`TryGetRepeatedArgument` is now separated from `TryGetArgument`. This allows this function to be simpler and avoid unnecessary instantiations of exoctic types - namely, `Tensor<std::vector<S>>`.